### ISSUE TITLE
NullReferenceException after recycle

### ISSUE THREAD

**Issue by jon-glass** [mikeedwards83/Glass.Mapper#137]
Hi,

We're experiencing an issue where Glass.Mapper throws a NullReferenceException if requests come in immediately after a recycle. It appears to occur if multiple requests come in while the recycled Sitecore app pool is still in the process of starting up and building out the cache. We're able to replicate it by sending around 10 requests within a few seconds of the recycle firing off. It usually fixes itself within a short time (5-10 seconds), though sometimes it will continue to throw exceptions for every request until another recycle is performed, like the context is corrupted.

We're using Glass.Mapper.Sc version 3.2.3.46. We've also tested upgrading to v4.0.1.5 locally, but the issue still exists, albeit with a different exception: 'Collection was modified; enumeration operation may not execute'.

Any insight or help would be greatly appreciated! Let me know if more detail is needed.

Thanks!

```
Exception Details: System.NullReferenceException: Object reference not set to an instance of an object.

Source Error: (Line 86)


Line 84:             //Bind country list with flags
Line 85:             var service = new SitecoreService(Sitecore.Context.Database);
Line 86:             var regionObj = service.QuerySingle<Continent>(string.Format(Queries.RegionByTitle, region.Replace("-", " ")));
Line 87:             if (regionObj == null)
Line 88:             {
```
```
[NullReferenceException: Object reference not set to an instance of an object.]
   Glass.Mapper.Configuration.<>c__DisplayClassb.<CreatePropertyExpression>b__a(Object obj, AbstractDataMappingContext context) +113

[MapperException: Failed to map property Date on SJM.Global.Framework.Models.Manuals.Continent]
   Glass.Mapper.Configuration.<>c__DisplayClassb.<CreatePropertyExpression>b__a(Object obj, AbstractDataMappingContext context) +545
   Glass.Mapper.Configuration.AbstractTypeConfiguration.MapPropertiesToObject(Object obj, IAbstractService service, AbstractTypeCreationContext context) +189

[MapperException: Failed to map properties on /sitecore/content/Manuals/Data/Regions/Europe.]
   Glass.Mapper.Configuration.AbstractTypeConfiguration.MapPropertiesToObject(Object obj, IAbstractService service, AbstractTypeCreationContext context) +394
   Glass.Mapper.Pipelines.ObjectConstruction.Tasks.CreateConcrete.CreateConcreteTask.CreateObject(ObjectConstructionArgs args) +968
   Glass.Mapper.Pipelines.ObjectConstruction.Tasks.CreateConcrete.CreateConcreteTask.Execute(ObjectConstructionArgs args) +588
   Glass.Mapper.Pipelines.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args) +197
   Glass.Mapper.Pipelines.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args) +293
   Glass.Mapper.Pipelines.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args) +293
   Glass.Mapper.Pipelines.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args) +293
   Glass.Mapper.Pipelines.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args) +293
   Glass.Mapper.Pipelines.AbstractPipelineRunner`2.Run(T args) +81
   Glass.Mapper.AbstractService.InstantiateObject(AbstractTypeCreationContext abstractTypeCreationContext) +646
   Glass.Mapper.Sc.SitecoreService.CreateType(Type type, Item item, Boolean isLazy, Boolean inferType, Dictionary`2 parameters, Object[] constructorParameters) +754
   Glass.Mapper.Sc.SitecoreService.CreateType(Item item, Boolean isLazy, Boolean inferType) +215
   Glass.Mapper.Sc.SitecoreService.QuerySingle(String query, Boolean isLazy, Boolean inferType) +231
   SJM.Global.Manuals.Layouts.Manuals.CountrySelector.BindCountryData(String region) in 
...
   System.Web.UI.Control.LoadRecursive() +71
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Control.LoadRecursive() +190
   System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint) +3178
```
Version(s)
```
  <package id="Castle.Core" version="3.2.2" targetFramework="net45" />
  <package id="Castle.Windsor" version="3.2.0" targetFramework="net45" />
  <package id="Glass.Mapper" version="3.0.14.26" targetFramework="net45" />
  <package id="Glass.Mapper.Sc" version="3.2.3.46" targetFramework="net45" />
  <package id="Glass.Mapper.Sc.CastleWindsor" version="3.3.0.24" targetFramework="net45" />
  <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net45" />
  <package id="WebActivatorEx" version="2.0.6" targetFramework="net45" />
```

**Comment by mark-saad** [comment]
I'm running into the same issue using version 4.0.1.5 
I'm upgrading from v2 which didn't have the issue. I have to kill the process to be able to get correct mappings, but I get the same stack trace in the logs.

**Comment by jon-glass** [comment]
We also first started experiencing the issue after upgrading from v2 to v3; we didn't have any issues prior to that.

**Comment by mikeedwards83** [comment]
I am trying to replicate this but it is  a challenge.

**Comment by mikeedwards83** [comment]
When you check the Sitecore logs is this exception in the shutting down process or the starting process?

**Comment by jon-glass** [comment]
It took quite a while for us to determine how to replicate it as well, and even then it wasn't replicated 100% of the time.

Its in the startup process, usually right around when its wrapping up creating the caches for each site definition.

**Comment by mikeedwards83** [comment]
Do you have any steps you could share? I have tried touching the Web.config
and then issuing a load of requests but this seems OK.

**Comment by mark-saad** [comment]
I just found out that our problem was a duplicate glass entity (two glass entities with different names but the same sitecore template ID) glass mapper would use the correct one the first time, after an application pool recycles it gets confused and throws the above exception. When we were using Glass Mapper v2 it was using the correct entity every time. 
After I removed the old glass entity, this behavior stopped and it seems to be working fine. 
If that's the case with Jon, I think we'd better add some logging around this edge case to help people find out what their issue is more easily.

**Comment by jon-glass** [comment]
We go through the following steps in order to reproduce it:
1) Initiate a manual (or scheduled) IIS recycle
2) Wait five seconds and then send 10-12 requests within a 1-2 second time span
3) Check the logs

Mark - I don't think we have an issue with duplicate Glass classes, but do you have any more details on your issue or how you discovered that was it? I could see if there are any similar scenarios on our end.

**Comment by mikeedwards83** [comment]
It looks like there might be two issues. Mark could you detail more about the duplicate glass entities? Are you using inferred types?

**Comment by mark-saad** [comment]
I'm using glass entities that were generated using the good old custom item generator, I didn't regenerate them recently because in my case you have to generate them one by one.
I had a list of SitecoreItem (parent type) and I use inferType to get the actual type of each item in the list, these items are then mapped to a view model.
At the first run, the list comes with the correct inferred types and they are mapped correctly. After recycle I found out that the first item gets mapped correctly but the second item throws an error because that second item was all nulls inside.
I went to look at the glass entity, it seemed fine, I searched for the template guid in the entire solution and then I found the second entity. I removed one of them and that was it.

Please let me know if I can be of any further help.

**Comment by benverhees** [comment]
We are having the same issue as jon-glass, tested with 4.0.1.5 and 4.0.1.8.
To reproduce locally, keep recycling your apppool while visiting the site in a couple of browsers. At some point, after 3 to 10 recycles, the error will appear.

**Comment by benverhees** [comment]
Testing after rolling back to version 3.3.1.51 shows that this also appears during recycling, but it is less visible because once the application has fully recycled, the error disappears.

**Comment by jon-glass** [comment]
@benverhees I have not found a suitable solution yet. Have you?

**Comment by mikeedwards83** [comment]
I am still struggling to replicate this problem. I am restarting the site by touching the web.config and firing off requests in fiddler.

**Comment by mikeedwards83** [comment]
Ok I have managed to replicate it :-)

**Comment by jon-glass** [comment]
Hi Mike,

That's great news. Have you figured out what the root issue is at all, and if so, is there a plan to update it in a future release? 

Thanks
Jon

**Comment by mikeedwards83** [comment]
Jon

I added additional logging after I replicated the issue and the problem stopped occurring. So now I am back to struggling to replicate.

Mike

**Comment by vince-geekcore** [comment]
It seems like we're also experiencing the same behavior on GlassMapper v4. When accessing pages based on models sometimes the following error is returned after a App Pool recycle.  

```
[NullReferenceException: Object reference not set to an instance of an object.]
   Glass.Mapper.IoC.AbstractConfigFactory`1.<GetItems>b__0(Func`1 f) +0
   System.Linq.WhereSelectListIterator`2.MoveNext() +243
   System.Linq.Buffer`1..ctor(IEnumerable`1 source) +487
   System.Linq.<ReverseIterator>d__a0`1.MoveNext() +140
   Glass.Mapper.Pipelines.AbstractPipelineRunner`2..ctor(IEnumerable`1 tasks) +503
   Glass.Mapper.AbstractService..ctor(Context glassContext) +113
   Glass.Mapper.Sc.Web.Ui.AbstractGlassUserControl..ctor() +166
   __ASP.FastObjectFactory_app_web_tit2yww3.Create_ASP_sublayouts_xxxxx_training_pijler_ascx() +48
   System.Web.UI.TemplateControl.LoadControl(IWebObjectFactory objectFactory, VirtualPath virtualPath, Type t, Object[] parameters) +232
   System.Web.UI.TemplateControl.LoadControl(VirtualPath virtualPath) +204
   Sitecore.Web.UI.WebControls.Sublayout.GetUserControl(Page page) +244
   Sitecore.Web.UI.WebControls.Sublayout.CreateChildControls() +331
   System.Web.UI.Control.EnsureChildControls() +188
   Sitecore.Layouts.Expander.Expand(Control control) +68
   Sitecore.Nexus.Web.NexusPlaceholderApi.(Placeholder ) +605
   Sitecore.Nexus.Web.NexusPlaceholderApi.CreateChildControls(Placeholder placeholder) +486
   System.Web.UI.Control.EnsureChildControls() +188
   Sitecore.Layouts.Expander.Expand(Control control) +68
   Sitecore.Layouts.Expander.Expand(Control control) +289
   Sitecore.Web.UI.WebControls.Sublayout.CreateChildControls() +455
   System.Web.UI.Control.EnsureChildControls() +188
   Sitecore.Layouts.Expander.Expand(Control control) +68
   Sitecore.Nexus.Web.NexusPlaceholderApi.(Placeholder ) +605
   Sitecore.Nexus.Web.NexusPlaceholderApi.CreateChildControls(Placeholder placeholder) +486
   System.Web.UI.Control.EnsureChildControls() +188
   Sitecore.Layouts.Expander.Expand(Control control) +68
   Sitecore.Layouts.Expander.Expand(Control control) +289
   Sitecore.Layouts.Expander.Expand(Control control) +289
   Sitecore.Layouts.PageContext.Build() +47
   (Object , Object[] ) +82
   Sitecore.Pipelines.CorePipeline.Run(PipelineArgs args) +364
   Sitecore.Layouts.PageContext.BuildControlTree(Object sender, EventArgs e) +210
   System.Web.UI.Page.PerformPreInit() +48
   System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint) +1843

```

```
Exception: System.IndexOutOfRangeException
Message: Index was outside the bounds of the array.
Source: mscorlib
   at System.Collections.Generic.List`1.Add(T item)
   at Glass.Mapper.Sc.IoC.ObjectConstructionTaskConfigFactory.AddTypes()
   at Glass.Mapper.IoC.AbstractConfigFactory`1.GetItems()
   at Glass.Mapper.AbstractService..ctor(Context glassContext)
   at Glass.Mapper.Sc.Web.Ui.AbstractGlassUserControl..ctor()
   at __ASP.FastObjectFactory_app_web_tit2yww3.Create_ASP_sublayouts_xxxxx_2015_training_pijler_ascx()
   at System.Web.UI.TemplateControl.LoadControl(IWebObjectFactory objectFactory, VirtualPath virtualPath, Type t, Object[] parameters)
   at System.Web.UI.TemplateControl.LoadControl(VirtualPath virtualPath)
   at Sitecore.Web.UI.WebControls.Sublayout.GetUserControl(Page page)
   at Sitecore.Web.UI.WebControls.Sublayout.CreateChildControls()
   at System.Web.UI.Control.EnsureChildControls()
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Nexus.Web.NexusPlaceholderApi.(Placeholder )
   at Sitecore.Nexus.Web.NexusPlaceholderApi.CreateChildControls(Placeholder placeholder)
   at System.Web.UI.Control.EnsureChildControls()
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Web.UI.WebControls.Sublayout.CreateChildControls()
   at System.Web.UI.Control.EnsureChildControls()
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Nexus.Web.NexusPlaceholderApi.(Placeholder )
   at Sitecore.Nexus.Web.NexusPlaceholderApi.CreateChildControls(Placeholder placeholder)
   at System.Web.UI.Control.EnsureChildControls()
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Layouts.PageContext.Expander.Expand(Control control)
   at Sitecore.Layouts.PageContext.Build()
   at (Object , Object[] )
   at Sitecore.Pipelines.CorePipeline.Run(PipelineArgs args)
   at Sitecore.Layouts.PageContext.BuildControlTree(Object sender, EventArgs e)
   at System.Web.UI.Page.PerformPreInit()
   at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
```

**Comment by andrew-at-sitecore** [comment]
We've run into quite similar situation. There seems to be two distinct stack traces associated with this situation:

<pre>[NullReferenceException: Object reference not set to an instance of an object.]
   Glass.Mapper.Configuration.<>c__DisplayClassb.< CreatePropertyExpression >b__a(Object obj, AbstractDataMappingContext context) +72

[MapperException: Failed to map property SortOrder on _CustomNamespace_.Web.Models.Base.BaseItem]
   Glass.Mapper.Configuration.<>c__DisplayClassb.< CreatePropertyExpression >b__a(Object obj, AbstractDataMappingContext context) +391
   Glass.Mapper.Configuration.AbstractTypeConfiguration.MapPropertiesToObject(Object obj, IAbstractService service, AbstractTypeCreationContext context) +91

[MapperException: Failed to map properties on /sitecore/content/_SiteRoot_/Home.]
   Glass.Mapper.Configuration.AbstractTypeConfiguration.MapPropertiesToObject(Object obj, IAbstractService service, AbstractTypeCreationContext context) +268
   Glass.Mapper.Pipelines.ObjectConstruction.Tasks.CreateConcrete.CreateConcreteTask.CreateObject(ObjectConstructionArgs args) +358

[MapperException: Failed to create type _CustomNamespace_.Web.Models.DataTemplates.Base.BasePage]
   Glass.Mapper.Pipelines.ObjectConstruction.Tasks.CreateConcrete.CreateConcreteTask.CreateObject(ObjectConstructionArgs args) +564
   Glass.Mapper.Pipelines.ObjectConstruction.Tasks.CreateConcrete.CreateConcreteTask.Execute(ObjectConstructionArgs args) +248
   Glass.Mapper.Pipelines.<>c__DisplayClass3.< CreateTaskExpression >b__2(T args) +70
   Glass.Mapper.Pipelines.<>c__DisplayClass3.< CreateTaskExpression >b__2(T args) +92
   Glass.Mapper.Pipelines.<>c__DisplayClass3.< CreateTaskExpression >b__2(T args) +92
   Glass.Mapper.Pipelines.<>c__DisplayClass3.< CreateTaskExpression >b__2(T args) +92
   Glass.Mapper.Pipelines.<>c__DisplayClass3.< CreateTaskExpression >b__2(T args) +92
   Glass.Mapper.Pipelines.<>c__DisplayClass3.< CreateTaskExpression >b__2(T args) +92
   Glass.Mapper.AbstractService.InstantiateObject(AbstractTypeCreationContext abstractTypeCreationContext) +205
   Glass.Mapper.Sc.SitecoreService.CreateType(Type type, Item item, Boolean isLazy, Boolean inferType, Dictionary`2 parameters, Object[] constructorParameters) +311
   Glass.Mapper.Sc.SitecoreService.GetItem(String path, Boolean isLazy, Boolean inferType) +134</pre>

<pre>
[NullReferenceException: Object reference not set to an instance of an object.]
   Glass.Mapper.IoC.AbstractConfigFactory`1.< GetItems >b__0(Func`1 f) +0
   System.Linq.WhereSelectListIterator`2.MoveNext() +244
   System.Linq.Buffer`1..ctor(IEnumerable`1 source) +519
   System.Linq.<ReverseIterator>d__a0`1.MoveNext() +140
   Glass.Mapper.Pipelines.AbstractPipelineRunner`2..ctor(IEnumerable`1 tasks) +503
   Glass.Mapper.AbstractService..ctor(Context glassContext) +197
   Glass.Mapper.Sc.SitecoreService..ctor(Database database, String contextName) +54
   lambda_method(Closure , Object[] ) +59
   Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate() +334

[DependencyResolutionException: An exception was thrown while invoking the constructor 'Void .ctor()' on type 'SitecoreContext'. ---> Object reference not set to an instance of an object. (See inner exception for details.)]
   Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate() +888
   Autofac.Core.Activators.Reflection.ReflectionActivator.ActivateInstance(IComponentContext context, IEnumerable`1 parameters) +361
   Autofac.Core.Resolving.InstanceLookup.Activate(IEnumerable`1 parameters) +89
   Autofac.Core.Resolving.InstanceLookup.Execute() +181
   Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, IComponentRegistration registration, IEnumerable`1 parameters) +168
   Autofac.ResolutionExtensions.ResolveService(IComponentContext context, Service service, IEnumerable`1 parameters) +168
   Autofac.ResolutionExtensions.Resolve(IComponentContext context, IEnumerable`1 parameters) +117
   _CustomNamespace_.Web.App_Start.<>c.< RenderDependencies >b__0_0(IComponentContext c) +41
   Autofac.Builder.<>c__DisplayClass1`1.< ForDelegate >b__0(IComponentContext c, IEnumerable`1 p) +15
   Autofac.Core.Activators.Delegate.DelegateActivator.ActivateInstance(IComponentContext context, IEnumerable`1 parameters) +32
   Autofac.Core.Resolving.InstanceLookup.Activate(IEnumerable`1 parameters) +89
   Autofac.Core.Resolving.InstanceLookup.Execute() +181
   Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, IComponentRegistration registration, IEnumerable`1 parameters) +168
   Autofac.Core.Resolving.ResolveOperation.Execute(IComponentRegistration registration, IEnumerable`1 parameters) +193
   Autofac.ResolutionExtensions.ResolveOptionalService(IComponentContext context, Service service, IEnumerable`1 parameters) +168
   System.Web.Mvc.DependencyResolverExtensions.GetService(IDependencyResolver resolver) +61
   _CustomNamespace_.Web.Controllers.BaseController..ctor() +42
   lambda_method(Closure , Object[] ) +60
   Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate() +335
</pre>

**Comment by mikeedwards83** [comment]
I have pushed an update to the develop branch (version 4.0.3). This should contain a fix for this issue. Please download the nightly pre-release build and let me know if this solves the issue.

**Comment by vince-geekcore** [comment]
I've downloaded the nightly release from the prerelease Nuget source. Checked the version of the dll in my bin folder (4.0.3.43). Unfortunately i was able to reproduce the errors after a couple of App Pool recycles. Looks like the same problem:

```
Exception: System.IndexOutOfRangeException
Message: Index was outside the bounds of the array.
Source: mscorlib
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Glass.Mapper.Sc.Pipelines.ObjectConstruction.EnforcedTemplateCheck.Execute(ObjectConstructionArgs args)
   at Glass.Mapper.Pipelines.AbstractPipelineRunner`2.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args)
   at Glass.Mapper.Pipelines.AbstractPipelineRunner`2.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args)
   at Glass.Mapper.Pipelines.AbstractPipelineRunner`2.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args)
   at Glass.Mapper.Pipelines.AbstractPipelineRunner`2.<>c__DisplayClass3.<CreateTaskExpression>b__2(T args)
   at Glass.Mapper.AbstractService.InstantiateObject(AbstractTypeCreationContext abstractTypeCreationContext)
   at Glass.Mapper.Sc.SitecoreService.CreateType(Type type, Item item, Boolean isLazy, Boolean inferType, Dictionary`2 parameters, Object[] constructorParameters)
   at Glass.Mapper.Sc.SitecoreService.CreateType[T](Item item, Boolean isLazy, Boolean inferType, Object[] constructorParameters)
   at Glass.Mapper.Sc.SitecoreService.CreateType[T](Item item, Boolean isLazy, Boolean inferType)
   at Glass.Mapper.Sc.Web.Ui.GlassUserControl`1.GetModel()
   at Glass.Mapper.Sc.Web.Ui.GlassUserControl`1.OnLoad(EventArgs e)
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Control.LoadRecursive()
   at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint)
```

**Comment by mikeedwards83** [comment]
Hi
This is a slightly different stack trace. I have just pushed a fix to dev, version 4.0.3.44.

Mike

**Comment by vince-geekcore** [comment]
Thanks Mike, all seems good for now!  Will test it some more.

**Comment by seankearney** [comment]
We were seeing similar stack traces in multiple projects. Restarting the web app was the only resolution. This stack trace was from 4.0.1.5

```
Exception: System.NullReferenceException
Message: Object reference not set to an instance of an object.
Source: Glass.Mapper
   at Glass.Mapper.IoC.AbstractConfigFactory`1.<GetItems>b__0(Func`1 f)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.<ReverseIterator>d__a0`1.MoveNext()
   at Glass.Mapper.Pipelines.AbstractPipelineRunner`2..ctor(IEnumerable`1 tasks)
   at Glass.Mapper.AbstractService..ctor(Context glassContext)
   at Glass.Mapper.Sc.ItemExtensions.GlassCast[T](Item item, Boolean isLazy, Boolean inferType)
   at ASP._Page_Areas_XXXXX_Views_Layouts_DefaultLayout_cshtml.Execute() in c:\inetpub\XXXXX\Website\Areas\XXXXX\Views\Layouts\DefaultLayout.cshtml:line 78
   at System.Web.WebPages.WebPageBase.ExecutePageHierarchy()
   at System.Web.Mvc.WebViewPage.ExecutePageHierarchy()
   at System.Web.WebPages.WebPageBase.ExecutePageHierarchy(WebPageContext pageContext, TextWriter writer, WebPageRenderingBase startPage)
   at System.Web.Mvc.Html.PartialExtensions.Partial(HtmlHelper htmlHelper, String partialViewName, Object model, ViewDataDictionary viewData)
   at Sitecore.Mvc.Presentation.ViewRenderer.Render(TextWriter writer)
```

I've upgraded one of the solutions to 4.0.3.44-beta and will be testing and (ideally pushing to production).

**Comment by seankearney** [comment]
@vince-geekcore Can you provide an update on your testing over the last week?

**Comment by mikeedwards83** [comment]
@seankearney Does this happen after a recycle? Does this affect all requests to the server?

**Comment by vince-geekcore** [comment]
@seankearney testing has been going smoothly. We can't seem to reproduce the issue anymore and did not encounter any negative side effects. Really glad this seems to be fixed.

**Comment by seankearney** [comment]
@mikeedwards83 we saw this three times over five days. Each time was the result of the app pool recycling "because the worker process reached its allowed processing time limit." So the app pool recycled, then immediately went into this state. It takes another recycle to get the site responding again. I have a detailed analysis of the problem if you would like it.

**Comment by seankearney** [comment]
Thanks @vince-geekcore! Sounds promising!

**Comment by cardinal252** [comment]
@seankearney - are you using the config factories in the out of the box configuration ?? previous stack traces showed Autofac in use ?

**Comment by mikeedwards83** [comment]
@seankearney @cardinal252 I have moved this issue to https://github.com/mikeedwards83/Glass.Mapper/issues/172

It doesn't look like it is related to the earlier issues since it originates in a different location.

**Comment by mikeedwards83** [comment]
This issue has been fixed in release 4.0.3.50
